### ISSUE TITLE
Numeric value does not handle values too low or too high for type TValue

### ISSUE THREAD

**Issue by InzeNL** [Megabit/Blazorise#2876]
**Describe the bug**
When the value of a numeric input is higher or lower than the `TValue` can handle, it does not throw an error. Instead, it keeps the last valid value it had.

**To Reproduce**
1. Clone my repository https://github.com/InzeNL/Blazorise-NumericEdit-Bug-Demo
2. Launch the application
3. Enter 0 as a value
4. The div above the input should now show "0"
5. Remove the value
6. The div above the input should now show nothing
7. Enter 1 as a value
8. The div above the input should now show "1"
9. Add a "-" in front of the 1
10. The div above the input still shows "1"
11. Enter 1111111111 as a value
12. The div above the input should now show "1111111111"
13. Add another number to the end of this value
14. The div above the input still shows "1111111111"

**Expected behavior**
I am not sure what the best behaviour would be, but I think the most reasonable would be as follows:
If the value (e.g. -1, or 11111111112) does not fit in the value type (e.g. uint), set the value to "null"

This makes it possible to verify that the values are valid.
Right now, if I enter a valid value, and then enter a value too big, it uses the last valid value.
The user does not see any feedback, and thinks "11111111112" is a valid value. The verification thinks the value is actually "1111111111", and uses that.

**Screenshots**
![image](https://user-images.githubusercontent.com/2505862/132332540-dfd22ab8-438c-4351-a0e0-77c9d60f6ba1.png)
![image](https://user-images.githubusercontent.com/2505862/132332568-3d5857d5-54b1-43e2-903e-aa7ad21db950.png)
![image](https://user-images.githubusercontent.com/2505862/132332585-e0df7fc1-7c4a-4e69-b3b7-d6265f376fe1.png)
![image](https://user-images.githubusercontent.com/2505862/132332609-869511c2-1887-4313-bd52-738bcb0ece6b.png)
![image](https://user-images.githubusercontent.com/2505862/132332642-8b063554-e948-4381-a0f1-d10d34ea6644.png)
![image](https://user-images.githubusercontent.com/2505862/132332678-5e968e8c-9e0e-4ecc-8a31-c3cf134ed4f0.png)

**Comment by InzeNL** [comment]
Looking at the code, the value converter already returns an empty value
https://github.com/Megabit/Blazorise/blob/b55e31f307cd8d342cb3ba829220db42d650562d/Source/Blazorise/Components/NumericEdit/NumericEdit.razor.cs#L147

In the value handler, it checks whether it was successfully parsed. Perhaps there should be an `else` that resets the input value to the `CurrentValue`? That way, the user gets feedback that the value they tried to enter was invalid.
https://github.com/Megabit/Blazorise/blob/b55e31f307cd8d342cb3ba829220db42d650562d/Source/Blazorise/Base/BaseInputComponent.razor.cs#L143

**Comment by CaiusJard** [comment]
As a workaround can you set the min and max properties of the edit, to the T.MinValue/T.MaxValue?

**Comment by InzeNL** [comment]
Correct, that is what I am using as a workaround in my current application.
However, I also use validation attributes, meaning I now define the Min and Max in two places.
Not per se any rush behind this, but would be cleaner to handle it in a different way

**Comment by InzeNL** [comment]
I guess this could be fixed by setting the default Min and Max values of a NumericEdit to T.MinValue and T.MaxValue on initialization

**Comment by stsrki** [comment]
@InzeNL Sorry for not answering sooner. Yes, I think this is the bug. Also while testing I did notice another bug with the step-down command. I will try to fix it for the next update.

**Comment by stsrki** [comment]
Note for me:

**The fix for step-down**

```cs
protected virtual TValue AddStep( TValue value, int sign )
{
    // make sure that null values also starts from zero
    value ??= Converters.ChangeType<TValue>( 0 );

    return sign > 0
        ? MathUtils<TValue>.Add( value, Converters.ChangeType<TValue>( Step.GetValueOrDefault( 1 ) ) )
        : MathUtils<TValue>.Subtract( value, Converters.ChangeType<TValue>( Step.GetValueOrDefault( 1 ) ) );
}
```

**Comment by StefanOssendorf** [comment]
That won't fix the display issue. That's exactly the same issue as I had in the gitter discussion with you.
You first need to let the invalid value render through Blazor and then set it back to a valid one. Otherwise Blazor does not pick up the changed value and the UI representation and the actual bound value start to diverge.
This is your code from our gitter discussion:
```csharp
 async Task TimePicked( string value )
{
    var old = _inputValue;
    _inputValue = value;

    if ( CheckValueHere )
    {
        await InvokeAsync( StateHasChanged );

        await Task.Yield();

        _inputValue = old;
    }
}
```

**Comment by stsrki** [comment]
I have investigated more and there is in fact a reason why it works like this. NumericEdit is supposed to let you write whatever value you want to, even larger than it supports. But then, **only** when you lost focus it will check for the Min and Max and then readjust the value. I have re-checked with what I did in the past and compared it to the other frameworks and they all behave the same. If the value is outside of Min and Max it will readjust on focus lost event.

To be honest I could easily change the behavior and prevent value while it is written. But by doing so it feels even weirder. For example, if you assign the TValue as `short`, its Max is 32768, and you start writing `45678`, it will allow you only up to `4567` and `8` will be prevented. Again very different from other frameworks.

I think changing the behavior would be breaking change now. If I would to make it, it would need to go to 0.9.5 release.
